### PR TITLE
STM32 MSC Support for SDIO boards -> SKR 2

### PR DESCRIPTION
**Pull request by rhapsodyv**
### Description

This PR add support for SDIO board to use Mass Storage Class (when the board has native USB support).

This the case of SKR 2. This adds a `_USB` env for SKR 2 to enable MSC for its onboard SDIO SD.

### Requirements

SKR 2

### Benefits

SDIO boards now can use MSC
SKR 2 with MSC

### CODE REVIEW

**Comment by thisiskeithb** [comment]
What are your thoughts on merging `*_USB` envs for chips without enough room (like this one)? We enable MSC on LPC for example without having extra environments.

**Comment by looxonline** [comment]
I believe that this will also affect the octopus. Currently I need to disable SDIO in order to use the USB env. I was actually going to submit a PR which disables it for the USB env but if this will fix it then there is no need (although there is also another build flag that is broken that I will probably submit for).

**Comment by Rikrdoga** [comment]
i just test it and it's working, thank u

File: ini/stm32f4.ini [hunk 1]
```diff
@@ -243,6 +243,15 @@ build_flags       = ${stm_flash_drive.build_flags}
   -DUSE_USBHOST_HS -DUSE_USB_HS_IN_FS -DUSBD_IRQ_PRIO=5 -DUSBD_IRQ_SUBPRIO=6
   -DHSE_VALUE=8000000U -DHAL_SD_MODULE_ENABLED
 
+#
+# BigTreeTech SKR V2.0 (STM32F407VGT6 ARM Cortex-M4) with USB Media Share Support
+#
+[env:BIGTREE_SKR_2_USB]
+extends           = env:BIGTREE_SKR_2
```
Comment (thinkyhead) [inline ini/stm32f4.ini:251]: Note: The PlatformIO IDE extension for VSCode requires a `platform` field or it won't display the environment (last time I checked).

**Comment by gordo3di** [comment]
For some reason I can see the LCD SD card through windows explorer but not onboard via USB. I tried MULTI_VOLUME as well. Does this work on the SKR v2 for everyone else? I was hoping to see both and if I set the SDCARD_CONNECTION to ONBOARD it will work in touchscreen mode but then you can't access the LCD SD if using marlin mode.

 ```
 #define SDCARD_CONNECTION LCD

  // Enable if SD detect is rendered useless (e.g., by using an SD extender)
  #define NO_SD_DETECT

  // Multiple volume support - EXPERIMENTAL.
  #define MULTI_VOLUME
  #if ENABLED(MULTI_VOLUME)
    #define VOLUME_SD_ONBOARD
    //#define VOLUME_USB_FLASH_DRIVE
    #define DEFAULT_VOLUME SV_SD_ONBOARD
    #define DEFAULT_SHARED_VOLUME SV_SD_ONBOARD
    //#define DEFAULT_SHARED_VOLUME SV_USB_FLASH_DRIVE
  #endif
```

**Comment by rhapsodyv** [comment]
I never did any test using LCD SD, only the onboard one.

And multi volume doesn't work with LCD SD yet.

I will work on this soon.

**Comment by gordo3di** [comment]
> I never did any test using LCD SD, only the onboard one.
> 
> And multi volume doesn't work with LCD SD yet.
> 
> I will work on this soon.

Thanks. For now I ended up switching it to onboard since I figure it would make it easier for my customers to update firmware and they can still access the LCD SD, Onboard and USB thumb drive from the LCD screen anyway.

**Pull request merged**